UI modif

Louis-Etienne opened the pull request:
Modification du thème du UI et du layout, pour éviter de saigner des yeux!

Louis-Etienne committed:
new Layout just dropped

Louis-Etienne committed:
custom Window Size hook

Louis-Etienne committed:
Resizing du layout  + Fonctionnalité des channels

Louis-Etienne committed:
Cleanup + comments

Louis-Etienne committed:
more little comments

Louis-Etienne commented:
Ça devrait être clean John, hésite-pas à me donner des améliorations à faire :))

PladsElsker commented on frontend/App.tsx:
```diff
@@ -1,40 +1,84 @@
+    const [isWindowWidthSmall, setIsWindowWidthSmall] = useState(false);
+    const WINDOWSMALLSIZE = 780;
+    
+    //Set the isWindowWidthSmall flag
```
Enlève le comment (usually meilleur de mettre dans une fonction pis de la caller si tu veux nommer des affaires)

PladsElsker commented on frontend/src/components/Utilities/CustomButton.tsx:
```diff
@@ -0,0 +1,77 @@
+
+});
+
+export default CustomButton;
```
Plus une suggestion, mais c'est souvent une bonne idée de finir le fichier avec un new-line (\n, enter, whatever) parce que git hub/language standards/whatever je sais pu pourquoi:
https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log

PladsElsker commented on frontend/src/components/rightPanel.tsx:
```diff
@@ -1,82 +1,148 @@
-        this.state = {};
+        
+        {/*imageProp     : contains 4 spaces for the 3 channels and the initial state  */}
+        {/*channel       : current index of the channel picker */}
```
comments, remove

PladsElsker commented on frontend/src/components/rightPanel.tsx:
```diff
@@ -1,82 +1,148 @@
+    /*loads the channel images*/
+    async componentDidMount(){
+        //OUCH
+        //Load l'image par defaut - ce n'est pas un path relatif ouch - react-native est bizarre
```
remove comments

PladsElsker commented on frontend/src/components/topBar.tsx:
```diff
@@ -1,52 +1,207 @@
+            <>
+                {/*weird transparent layout shenanigans to create a touchable 
+                    layout to unfocus from dropdown when clicked outside!
+                    Merci John pour l'idee!!*/}
```
Remove comments (╯°Д°)╯︵┻─┻ bref t'as compris lol

PladsElsker commented:
Le reste à l'air fine! Il va manquer de rajouter la bottom bar ou dequoi du genre pour mettre les contrôles de la simulation, mais sinon c'est fine. Good job 👍

PladsElsker commented:
Ah autre chose aussi, j'ai pas dit parce que honnêtement ça me tentait pas de rentrer dans les détails (pis honnêtement je trouve ça têteux), mais si jamais ça te tente, essaie d'avoir plus de consistence pour la quantitée de new-lines entre les différentes sections du code (imports | component | style, voir aussi la quantitée de new-lines entre les lignes dans le body des fonctions elles-mêmes, etc. Pas mal plus facile à lire pour moi en tout cas, mais comme je disais c'est vraiment un commentaire têteux)

Louis-Etienne committed:
code cleaning, spacing fixed!

J'ai fait un peu du ménage quant aux commentaires! (J'en ai gardé quelque-uns parce que je crois qu'ils sont pertinents pour la lisibilité)

Sinon, j'ai aussi fait des modifcations pour uniformiser le spacing entre les gros blocs de code d'un fichier (gros bloc  = spacing de deux \n)

Voilà voilà :))

Louis-Etienne commented:
Nouveau commit :)) Pour fix les commentaires et le spacing!

PladsElsker commented:
Tous les commentaires qui restent selon moi devraient être enlevés (sauf les headers of course). 

Le code qui leur correspond me semble assez clair et self-explanatory.

Louis-Etienne committed:
commentaires futiles NUKED

Tous les commentaires inutiles (tous les commentaires) ont été enlevés

Louis-Etienne commented:
Je viens de commit en enlevant tous les comments :))
